actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -629,36 +629,23 @@ def match(path, options = nil)
         #
         # This will generate the `exciting_path` and `exciting_url` helpers which can be
         # used to navigate to this mounted app.
-        def mount(app, options = nil)
-          if options
-            path = options.delete(:at)
-          elsif Hash === app
-            options = app
-            app, path = options.find { |k, _| k.respond_to?(:call) }
-            options.delete(app) if app
-            hash_key_app = true
-          end
+        def mount(app, as: nil, via: nil, at: nil, defaults: nil, constraints: nil, anchor: false, format: false, path: nil, internal: nil, **mapping, &block)
```

Comment:
what is `**mapping`?  A method call to something?

Reply:
Mapping is route mapping parameters like https://github.com/rails/rails/blob/9ba208c16835f4a174ae9fd385ebc18972d758a4/actionpack/lib/action_dispatch/routing.rb#L133. To be honest, I didn't know you could do this. We might want to deprecate this in favour of a keyword, but in my testing it didn't speed things up at all.

Reply:
Yeah actually I think this is just a shorthand for constraints: https://github.com/rails/rails/blob/9ba208c16835f4a174ae9fd385ebc18972d758a4/actionpack/test/controller/routing_test.rb#L567

Reply:
As far as I can tell, strings passed as mapper parameters are treated as route defaults, and regexes are constraints for route parameters. I vote to deprecate this since there's already keywords for this.

Reply:
I think I can try to deal with mapping deprecations in a future PR. I need to introduce keywords in resource calls too which could be done as a followup or now.